packages/@ember/-internals/metal/lib/tracked.ts

```diff
@@ -8,6 +8,102 @@ import { markObjectAsDirty, tagForProperty } from './tags';
 
 type Option<T> = T | null;
 
+let WARN_IN_AUTOTRACKING_TRANSACTION = false;
+let AUTOTRACKING_TRANSACTION: WeakMap<Tag, Error> | null = null;
+
+export let runInAutotrackingTransaction: (fn: () => void) => void;
+export let warnInAutotrackingTransaction: (fn: () => void) => void;
+export let assertPropertyNotTracked: (
+  tag: Tag,
+  obj: object,
+  keyName?: string,
+  forHardError?: boolean
+) => void;
+
+if (DEBUG) {
+  runInAutotrackingTransaction = (fn: () => void) => {
+    AUTOTRACKING_TRANSACTION = new WeakMap();
```

Comment:
Please add an error when this is `!== null` before we assign it. It should not happen, but its definitely bad if it does...

Reply:
Ended up wrapping `track()` in an assertion in DEBUG as well (turns out modifiers run outside of render, so we do need to do it), so we can actually have nested transactions. I made sure it would work though.